scaloid-common/src/main/st/org/scaloid/common/app.scala

```diff
@@ -96,7 +108,17 @@ trait SActivity extends Activity with SContext with TraitActivity[SActivity] wit
 
   protected override def onCreate(b: Bundle) {
     super.onCreate(b)
-    onCreateBodies.foreach(_ ())
+    AppHelpers.createBundle.withValue(Option(b)) {
+      onCreateBodies.foreach(_ ())
+    }
+  }
+
+  def onCreate(body: Option[Bundle] => Any) = {
+    val el = { () =>
+      body(AppHelpers.createBundle.value)
```

Comment:
The advantage of using a `DynamicVariable` is to be able to use it as an instance attribute, so that would enable to have this functionality without breaking the API by having to change the signature of the body. But here you are still changing the signature so there is no need for this `DynamicVariable`. In `onCreate` it's in scope and with the new signature you just have to pass it as argument.